### PR TITLE
zh-translation: content/zh/docs/ops/best-practices/traffic-management…

### PR DESCRIPTION
**Pull request by mossinfan**
I want to translate content/zh/docs/ops/best-practices/traffic-management/index.md  as Chinese.
Please help to review.
Thanks.

### CODE REVIEW

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 1]
```diff
@@ -9,30 +9,20 @@ aliases:
...
+本章节提供具体的部署或者配置指南，避免出现网络或者流量管理问题.
 
-## Set default routes for services
+## 为服务设置默认路由
 
-Although the default Istio behavior conveniently sends traffic from any
-source to all versions of a destination service without any rules being set,
-creating a `VirtualService` with a default route for every service,
-right from the start, is generally considered a best practice in Istio.
+尽管我们不用设置任何的规则，默认的Istio行为就可以方便的从任意源地址发送流量到目标服务的所有版本，但是为每一个服务创建一个带有默认路由的 `VirtualService` 从一开始就被普遍认为是Istio的一个最佳实践。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
尽管我们不用设置任何的规则，默认的 Istio 行为就可以方便的从任意源地址发送流量到目标服务的所有版本，但是为每一个服务创建一个带有默认路由的 `VirtualService` 从一开始就被普遍认为是 Istio 的一个最佳实践。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 2]
```diff
@@ -9,30 +9,20 @@ aliases:
...
+即使最初你的服务只有一个版本，但只要你决定部署第二个版本，你就需要在你的新版本服务启动**之前**，就创建一个合适的路由规则，以防止它在不受控制的情况下立即收到流量。
 
-Another potential issue when relying on Istio's default round-robin routing is
-due to a subtlety in Istio's destination rule evaluation algorithm.
-When routing a request, Envoy first evaluates route rules in virtual services
-to determine if a particular subset is being routed to.
-If so, only then will it activate any destination rule policies corresponding to the subset.
-Consequently, Istio only applies the policies you define for specific subsets if
-you **explicitly** routed traffic to the corresponding subset.
+依靠Istio默认轮询(round-robin)路由方式的另一个潜在问题是因为Istio的目标规则评估算法有些微妙。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
依靠 Istio 默认轮询（round-robin）路由方式的另一个潜在问题是因为 Istio 的目标规则评估算法有些微妙。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 3]
```diff
@@ -9,30 +9,20 @@ aliases:
...
+当路由请求时，Envoy首先评估虚拟服务中的路由规则，以确定一个特定的网格是否可以被路由到。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
当路由请求时，Envoy 首先评估虚拟服务中的路由规则，以确定一个特定的网格是否可以被路由到。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 4]
```diff
@@ -9,30 +9,20 @@ aliases:
...
+如果是，则它将仅仅激活与该网格相对应的所有目标规则策略。
+因此，如果你**明确**地把流量路由到了对应的subnet，那么Istio只应用你为特定网格定义的策略。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
因此，如果你**明确**地把流量路由到了对应的 subnet，那么 Istio 只应用你为特定网格定义的策略。
```

Comment (yuxiaobo96) [reply content/zh/docs/ops/best-practices/traffic-management/index.md]: 下同：英文与中文间加空格
下面翻译有类似问题，就不一一指出了

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 5]
```diff
@@ -93,14 +80,11 @@ spec:
...
-## Control configuration sharing across namespaces {#cross-namespace-configuration}
+## 跨命名空间中共享控制配置 {#cross-namespace-configuration}
 
-You can define virtual services, destination rules, or service entries
-in one namespace and then reuse them in other namespaces, if they are exported
-to those namespaces.
-Istio exports all traffic management resources to all namespaces by default,
-but you can override the visibility with the `exportTo` field.
-For example, only clients in the same namespace can use the following virtual service:
+你可以在一个命名空间中定义虚拟服务，目标规则或者服务入口，如果他们被输出到其他的命名空间，他们就能在其他命名空间被重用。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
你可以在一个命名空间中定义虚拟服务、目标规则或者服务入口，如果他们被输出到其他的命名空间，他们就能在其他命名空间被重用。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 6]
```diff
@@ -93,14 +80,11 @@ spec:
...
+默认情况下，Istio将所有流量管理资源到输出到所有命名空间，但你可以使用 `exportTo` 字段覆盖这个可见性。
+例如，只有相同命名空间下的客户端彩可以使用以下虚拟服务：
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
例如，只有相同命名空间下的客户端才可以使用以下虚拟服务：
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 7]
```diff
@@ -215,14 +181,11 @@ spec:
     ...
 {{< /text >}}
 
-The downside of this kind of configuration is that other configuration (e.g., route rules) for any of the
-underlying microservices, will need to also be included in this single configuration file, instead of
-in separate resources associated with, and potentially owned by, the individual service teams.
-See [Route rules have no effect on ingress gateway requests](/zh/docs/ops/common-problems/network-issues/#route-rules-have-no-effect-on-ingress-gateway-requests)
-for details.
+这种配置的缺点是任何基础微服务的其他配置(例如路由规则)，也将需要包含在此单配置文件中，而不是
+在各个相关独立的服务团队并可能由其拥有独立的资源中。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
由各个独立的服务团队相关联并可能由其独立拥有的资源中。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 8]
```diff
@@ -215,14 +181,11 @@ spec:
...
+你可以在 [路由规则对入口网关的请求没有影响](/zh/docs/ops/common-problems/network-issues/#route-rules-have-no-effect-on-ingress-gateway-requests)中查看更多信息。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
请参阅 [路由规则对入口网关的请求没有影响](/zh/docs/ops/common-problems/network-issues/#route-rules-have-no-effect-on-ingress-gateway-requests)获取更多信息。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 9]
```diff
@@ -265,57 +228,45 @@ metadata:
   name: myapp-...
 {{< /text >}}
 
-When a second and subsequent `VirtualService` for an existing host is applied, `istio-pilot` will merge
-the additional route rules into the existing configuration of the host. There are, however, several
-caveats with this feature that must be considered carefully when using it.
+当现有主机应用第二个和后续的 `VirtualService` 时， `istio-pilot` 将会合并新加入的路由规则进入现有的主机配置中。然而有几个在使用该特性时必须仔细考虑的注意事项。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
当现有主机应用第二个和后续的 `VirtualService` 时， `istio-pilot` 将新加入的路由规则合并到现有的主机配置中。然而在使用该特性时有几个必须仔细考虑的注意事项。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 10]
```diff
@@ -265,57 +228,45 @@ metadata:
...
+1. 一个 `VirtualService` 只有绑定到了一个网关，它才能被分割。
+   Sidecar不支持主机合并。
+1. `DestinationRule` 也可以用类似的合并语义和限制来分割。
+1. 对于同一主机，跨多个目标规则的任何给定的网格应该只有一个定义。
+   如果有多个具有相同名称的定义，则使用第一个定义，并且丢弃后面的任何副本。
+   不支持合并网格的内容。
+1. 对于同一个主机，应该只有一个顶级的 `trafficPolicy` 。
+   当在多个目标规则中定义顶级流量策略时，将使用第一个。
+   后面的任何顶级 `trafficPolicy` 配置都将被丢弃。
+1. 与虚拟服务合并不同，目标规则的合并可以同事在sidecar和网关中工作。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
1. 与虚拟服务合并不同，目标规则的合并可以同时在sidecar和网关中工作。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 11]
```diff
@@ -265,57 +228,45 @@ metadata:
...
+## 在重新配置服务路由时避免503错误
 
-1. There should only be one definition of any given subset across multiple destination rules for the same host.
-   If there is more than one with the same name, the first definition is used and any following duplicates are discarded.
-   No merging of subset content is supported.
-1. There should only be one top-level `trafficPolicy` for the same host.
-   When top-level traffic policies are defined in multiple destination rules, the first one will be used.
-   Any following top-level `trafficPolicy` configuration is discarded.
-1. Unlike virtual service merging, destination rule merging works in both sidecars and gateways.
+当设置路由规则将流量导向一个服务特定的版本（网格）时，必须确保这些网格在路由中被使用之前是可用的。否则在重新配置期间，对服务的调用可能会返回503错误。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
当设置路由规则将流量导向一个服务特定的版本（网格）时，必须确保这些网格在路由中被使用之前是可用的。否则在重新配置期间，对服务的调用可能会返回 503 错误。
```

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 12]
```diff
@@ -265,57 +228,45 @@ metadata:
...
 
-## Avoid 503 errors while reconfiguring service routes
+使用单个 `kubectl` 调用创建定义了相应网格的 `VirtualServices` 和 `DestinationRules` (例如 `kubectl apply -f myVirtualServiceAndDestinationRule.yaml` ) 是不充分的，因为资源以最终一致的方式传播（从配置服务，例如 Kubernetes API server）到Pilot实例。 如果使用网格的 `VirtualService` 比定义了网格的 `DestinationRule` 先到达，那么有Pilot生成的Envoy配置将会引用不存在的上游池。这将导致HTTP 503错误，直到对于Pilot，所有的配置对象都可使用。
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]:
```suggestion
使用单个 `kubectl` 调用创建定义相应网格的 `VirtualServices` 和 `DestinationRules`（例如 `kubectl apply -f myVirtualServiceAndDestinationRule.yaml`）是不充分的，因为资源以最终一致的方式传播（从配置服务，例如 Kubernetes API server）到 Pilot 实例。 如果使用网格的 `VirtualService` 定义网格的 `DestinationRule` 先到达，那么由 Pilot 生成的 Envoy 配置将会引用不存在的上游池。这将导致 HTTP 503 错误，直到对于 Pilot，所有的配置对象都可使用。
```

Comment (yuxiaobo96) [reply content/zh/docs/ops/best-practices/traffic-management/index.md]: 数字与中文间建议加空格

**Comment by mossinfan** [comment]
Hi:  Yuxiaobo
I have fixed the translate issue as suggest. 
Would you help to review again?
Thanks

File: content/zh/docs/ops/best-practices/traffic-management/index.md [hunk 13]
```diff
@@ -93,14 +80,11 @@ spec:
...
 
-You can define virtual services, destination rules, or service entries
-in one namespace and then reuse them in other namespaces, if they are exported
-to those namespaces.
-Istio exports all traffic management resources to all namespaces by default,
-but you can override the visibility with the `exportTo` field.
-For example, only clients in the same namespace can use the following virtual service:
+你可以在一个命名空间中定义虚拟服务、目标规则或者服务入口，如果他们被输出到其他的命名空间，他们就能在其他命名空间被重用。
+默认情况下，Istio 将所有流量管理资源到输出到所有命名空间，但你可以使用 `exportTo` 字段覆盖这个可见性。
+例如，只有相同命名空间下的客户端彩可以使用以下虚拟服务：
```
Comment (yuxiaobo96) [inline content/zh/docs/ops/best-practices/traffic-management/index.md]: 这里没有修改

Comment (yuxiaobo96) [reply content/zh/docs/ops/best-practices/traffic-management/index.md]: 87行：彩 -> 才

**Comment by tanjunchen** [comment]
@mossinfan Thanks your jobs!
first : please refer to the origin issue , so this can trigger the webhook to close origin issue.
second : squash your commits (git rebase)

**Comment by yuxiaobo96** [comment]
This document has been translated in other pr, so close this pr.
time: 2020-01-19 09:41:28

**Pull request closed without merging**